setup/src/main/java/io/jenkins/jenkinsfile/runner/util/JenkinsRecipe.java

```diff
@@ -0,0 +1,57 @@
+package io.jenkins.jenkinsfile.runner.util;
+
+import java.lang.annotation.Documented;
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import java.lang.annotation.Retention;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import java.lang.annotation.Target;
+import java.lang.annotation.Annotation;
+import java.io.File;
+
+import io.jenkins.jenkinsfile.runner.JenkinsRule;
+import junit.framework.TestCase;
+
+
+/**
+ * Meta-annotation for recipe annotations, which controls
+ * the test environment set up.
+ *
+ * @author Kohsuke Kawaguchi
+ * @since 1.436
+ */
+@Retention(RUNTIME)
+@Documented
+@Target(ANNOTATION_TYPE)
+public @interface JenkinsRecipe {
+    /**
+     * Specifies the class that sets up the test environment.
+     *
+     * <p>
+     * When a recipe annotation is placed on a test method,
+     */
+    Class<? extends Runner> value();
+
+    /**
+     * The code that implements the recipe semantics.
+     *
+     * @param <T>
+     *      The recipe annotation associated with this runner.
+     */
+    abstract class Runner<T extends Annotation> {
```

Comment:
Not sure about compatibility, but would it be possible to refactor this into an interface with default methods?

Reply:
Would prefer to create a follow-up task for now